src/lib/protocols/credentials/CredentialService.ts

```diff
@@ -103,7 +103,11 @@ export class CredentialService extends EventEmitter {
     credDef: CredDef
   ): Promise<CredentialRequestMessage> {
     const proverDid = connection.did;
-    const offer = MessageTransformer.toMessageInstance(credential.offer, CredentialOfferMessage);
+
+    // FIXME: credential.offer is already CredentialOfferMessage type
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const offer = JsonTransformer.fromJSON(credential.offer, CredentialOfferMessage);
```

Comment:
Should've made this more clear. TS thinks it is already `CredentialOfferMessage`, however it still needs to be transformed. This can be confusing as you think it is already transformed, but it is not. See the suggested change

```suggestion
    // FIXME: TypeScript thinks the type of credential.offer is already CredentialOfferMessage, but it still needs to be transformed
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const offer = JsonTransformer.fromJSON(credential.offer, CredentialOfferMessage);
```

Reply:
I see. Yes, that's better 👍